src/engine/sparqlExpressions/BlankNodeExpression.cpp

```diff
@@ -0,0 +1,106 @@
+//   Copyright 2025, University of Freiburg,
+//   Chair of Algorithms and Data Structures.
+//   Author: Robin Textor-Falconi <textorr@informatik.uni-freiburg.de>
+
+#include "engine/sparqlExpressions/BlankNodeExpression.h"
+
+#include <absl/strings/str_cat.h>
+
+#include "engine/sparqlExpressions/LiteralExpression.h"
+#include "engine/sparqlExpressions/NaryExpressionImpl.h"
+#include "engine/sparqlExpressions/SparqlExpressionValueGetters.h"
+#include "util/ChunkedForLoop.h"
+
+namespace sparqlExpression {
+
+namespace detail {
+
+// Turn a blank node string into an iri with a special prefix.
+ad_utility::triple_component::Iri createBlankNodeForLabel(
+    std::string_view label) {
+  AD_CONTRACT_CHECK(label.starts_with("_:"));
+  std::string iriString =
+      absl::StrCat(QLEVER_INTERNAL_BLANK_NODE_IRI_PREFIX, label, ">");
+  return ad_utility::triple_component::Iri::fromStringRepresentation(
+      std::move(iriString));
+}
+
+// Escape a string for use as a blank node label.
+std::string escapeStringForBlankNode(std::string_view input) {
+  std::string output;
+  for (char c : input) {
+    // TODO<RobinTF> Implement actual escaping algorithm.
+    if ((c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z')) {
+      output.push_back(c);
+    }
+  }
+  return output;
+}
+
+// Map a string to a blank node.
+[[maybe_unused]] auto stringToBlankNodeImpl =
+    [](std::optional<std::string> s) -> IdOrLiteralOrIri {
+  if (s.has_value()) {
+    std::string label =
+        BlankNode{false, escapeStringForBlankNode(std::move(s.value()))}
+            .toSparql();
+    return LiteralOrIri{createBlankNodeForLabel(label)};
+  } else {
+    return Id::makeUndefined();
+  }
+};
+
+// SparqlExpression representing the term "BNODE("something")".
+using SpecificBlankNodeExpression =
```

Comment:
I am not sure, whether this expression also needs a `CacheBreaker`, but currently we don't care (write a TODO comment), because the semantics of the StringBlankNodes are really odd.